### PR TITLE
More unit testing

### PR DESCRIPTION
**Pull request by DanielElisenberg**
# MORE UNIT TESTING

* Added unit tests for the processor module. This module has thus far been mocked.
* Added error handling to processor for sensible HTTP responses
* unittest => pytest
* Some general clean up

### CODE REVIEW

File: tests/unit/core/test_processor.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    assert result_set_to_csv_string(file_name) == test_data.PERSON_INCOME_ALL
+
+
+def test_invalid_event_request():
+    with pytest.raises(HTTPException) as e:
+        processor.process_event_request(
+            test_data.INVALID_EVENT_QUERY_INVALID_STOP_DATE
+        )
+    assert "Empty result set" in e.value.detail
+    assert e.value.status_code == 204
```
Comment (pawbu) [inline tests/unit/core/test_processor.py:34]: Is this logic visible in OpenAPI docs?

File: tests/unit/core/test_processor.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    )
+
+
+def test_invalid_status_request():
+    with pytest.raises(HTTPException) as e:
+        processor.process_status_request(
+            test_data.INVALID_STATUS_QUERY_NOT_FOUND
+        )
+    assert "No such data structure" in e.value.detail
+    assert e.value.status_code == 404
```
Comment (pawbu) [inline tests/unit/core/test_processor.py:52]: Same here

File: tests/unit/core/test_processor.py (hunk 1 above)
Comment (DanielElisenberg) [reply tests/unit/core/test_processor.py:34]: It does not :/ Will try adding some app level exception handling that will autogenerate docs 🏃🏻

**Comment by DanielElisenberg** [comment]
@pawbu looked at [this hack](https://github.com/tiangolo/fastapi/issues/2455), but did not like it. Added a response_model module and explicitly mark each endpoint with additional possible responses.
Lifted error handling to the app configuration to catch self-defined exceptions from routes.

This is a little less utopian than the current autogenerated docs, but works pretty ok. Let me know what you think.

**Review by pawbu — APPROVED** [review]
Looks good!

**Pull request merged**